portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurrenceRule.java

```diff
@@ -0,0 +1,338 @@
+      builder.byRuleMap(byRuleMapInput);
+    }
+    return builder.build();
+  }
+
+  /**
+   * A method for parsing an RDate string into an {@link RDate}
+   *
+   * @param rDateString a string containing RDate information, excluding any prefixes (e.g.,
+   * RDATE:)
+   * @return an {@link RDate}
+   */
+  public static RDate parseRDateString(String rDateString) {
+    RDate.Builder builder = RDate.builder();
+    StringTokenizer stringTokenizer = new StringTokenizer(rDateString, ":");
+    String token = stringTokenizer.nextToken();
+    while (stringTokenizer.hasMoreTokens()) {
+      String[] split = token.split("=");
+      String key = split[0];
+      String value = split[1];
+      if (key.equals(VALUE)) {
+        builder.value(TimeType.valueOf(value));
+      } else if (key.equals(TZID)) {
+        builder.tzidparam(value);
+      }
+      token = stringTokenizer.nextToken();
+    }
+    builder.rdtval(Arrays.asList(token.split(",")));
+
+    return builder.build();
+  }
+
+  public static ExDate parseExDateString(String exDateString) {
+    ExDate.Builder builder = ExDate.builder();
+    StringTokenizer stringTokenizer = new StringTokenizer(exDateString, ":");
+    String token = stringTokenizer.nextToken();
+    while (stringTokenizer.hasMoreTokens()) {
+      String[] split = token.split("=");
+      String key = split[0];
+      String value = split[1];
+      if (key.equals(VALUE)) {
+        builder.value(TimeType.valueOf(value));
+      } else if (key.equals(TZID)) {
+        builder.tzidparam(value);
+      }
+      token = stringTokenizer.nextToken();
+    }
+    builder.exdtval(Arrays.asList(token.split(",")));
+
+    return builder.build();
+  }
+
+  @AutoValue
+  public abstract static class RRule {
+
+    abstract Freq freq();
+
+    @Nullable abstract String until();  // represents a date/time
+
+    @Nullable abstract Integer count();
+
+    @Nullable abstract Integer interval();
+
+    @Nullable abstract Map<ByRule, String> byRuleMap(); // TODO(olsona); parse properties further
+
+    @Nullable abstract Day wkst();
+
+    static RRule create(Freq freq, String until, int count, int interval,
+        Map<ByRule, String> byRuleMap, Day wkst) {
+      return builder()
+          .freq(freq)
+          .until(until)
+          .count(count)
+          .interval(interval)
+          .byRuleMap(byRuleMap)
+          .wkst(wkst)
+          .build();
+    }
+
+    static Builder builder() {
+      return new org.dataportabilityproject.types.transfer.models.calendar.AutoValue_RecurrenceRule_RRule.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      public abstract Builder freq(Freq freq);
+
+      public abstract Builder until(String until);
+
+      public abstract Builder count(Integer count);
+
+      public abstract Builder interval(Integer interval);
+
+      public abstract Builder byRuleMap(Map<ByRule, String> byRuleMap);
+
+      public abstract Builder wkst(Day wkst);
```

Comment:
What does this stand for? Let's rename or add comment.

Reply:
Added comment at the top about this.